### DISCUSSION TITLE
[24/04/26] 백엔드 1차 멘토링

### DISCUSSION THREAD

**Discussion by ghrltjdtprbs** [10aeat/10aeat_Backend#15] [백엔드 회의록]
### 답변 : 아님.

- **도메인 주도 설계**
    - Domain Driven Design
    - 엔티티를 도메인 기준으로 모아 둔다 ←
    - 도메인이란?
        - 테이블 묶는 하나의 비즈니스 관심사
- 예시

```java
- 주문 (도메인)
- 상품(도메인)
    - 상품(엔티티)
    - 재고(엔티티)
    - 상세설명(엔티티)
- 장바구니(도메인 == 엔티티 / 테이블)
- 결제(도메인) payment 패키지 → 모듈 → 서비스 → MSA
    - 포인트(도메인 == 엔티티 / 테이블) - 도메인 / 엔티티
        - point
            - pointService.java
            - pointRepository.java
            - point.java
    - 결제 내역 상세(엔티티)
- 회원(도메인)
```

**DDD에서의 도메인 구조**

1. 핵심
    - 어떠한 가치를 제공 할 것인지
    - 비즈니스 문제를 해결하기 위한 도메인
2. 지원
    - 지도
        - kakaomap
        - googlemap
    - 결제 외부 api 클라이언트 개발
        - NicePay
        - Toss
        - KakaoPay
        - NaverPay
3. 일반
    - 회원 가입 / 인증
    - 직접
    - 외주 / 외부 서비스 이용 가능
        - 간편 인증 (Oauth2)
        
- 질문 배경 : 패키지 구조 정하면서 고민
- 질문자 : 권재현

### 답변 :

**가장 먼저 해야할 일**

- Springboot application 하나 만들어서 배포 먼저 해두고 CD 구축해두기
- Nginx , server, db 등을 Dockerize 해서 관리하는 것 추천(도커는 현직에서도 많이 쓰임)

**추천**

- 서버 두 대 이상 띄우는 것 추천
→ 동시성 문제에 대해 고민해 볼 수 있습니다.
- 왜 무중단 배포 전략을 채택 했는지 이유를 잘 생각해볼 것

- 질문 배경 :  배포 아무도 해본 적 없음
- 질문자 : 최혜미

## 질문 3. wbs, convention 검토해주세요!

코드 컨벤션 : https://github.com/orgs/10aeat/discussions/3

WBS  :
https://docs.google.com/spreadsheets/d/11NsFaELkPpaq64GfFUuYtHKZZFtemUVT6RXW7rjFTwQ/edit#gid=558616111

### 답변 :

- WBS는 크게 신경쓰지 말 것. 요즘에 잘 안쓰는 방식(워터풀). 더군다나 백엔드는 깃 프로젝트를 활용해 스크럼 방식으로 개발할 예정이니 더 신경쓸 필요 x
- 컨벤션은 굿
→ 다만 패키지 구조의 global이 과연 global로 들어가도 되는지 고민해보면 좋을듯(팀원끼리 합의되면 상관x)

- 질문 배경 : 지금까지 한 것들이라서요..ㅠ
- 질문자 : 권재현

### 답변 :

- cd 구축
- 공통 클래스 설정(ex. ResponseDTO, exception … )
- 오피스너 앱 살펴보고 erd 설계 고민해보기
- 오피스너 입주자앱은 어떻게 구현됐는지 살펴보기

**Comment by asd42270** [comment]
저희 이번 주에 기획 나오기 전에 해야 할 것들 시작해야 하는데 혹시 괜찮으면 내일부터 시작하는 것은 어떠신지...제가 오늘은 학교 수업이 아침부터 저녁까지 풀로 있어섬..ㅠ

**Reply by Galmaeki** [reply]
저는 좋슴다!
시간을 조율해봐야할것같슴다!

**Reply by ghrltjdtprbs** [reply]
저도 좋습니다~
